### PR TITLE
🎨 UI revamp -  Theme & Table View

### LINKED ISSUE #685: Table circles look bad
Come up with and implement a better design.

### PR DESCRIPTION
**Pull request by gubo97000**
Closes #685

### CODE REVIEW

**Review by no-comma — COMMENTED** [review]
The UI looks good to me. On the code side, there are some ESLint warnings which should be fixed, and I also wonder what is the reason for leaving in so much commented out code?

File: client/src/components/Header.tsx [hunk 1]
```diff
@@ -106,30 +102,88 @@ const Header = (): JSX.Element => {
...
+                  variant="body1"
+                  sx={{color: 'primary.main'}}
+                >
+                  <b>{course.data.courseCode}</b>
+                </Typography>
+              </span>
+            </Box>
+            <Typography variant="h2" sx={{mr: 1}}>
+              {' '}
+              -{' '}
```
Comment (bntti) [inline client/src/components/Header.tsx:158]: Couldn't this just be `{' - '}`?

File: client/src/components/course-view/AttainmentsView.tsx [hunk 2]
```diff
@@ -275,27 +275,69 @@ const AttainmentsView = (): JSX.Element => {
...
             )}
-            <Button
-              onClick={handleSubmit}
-              variant={unsavedChanges ? 'contained' : 'text'}
-              disabled={error || editing}
-            >
-              Save
-            </Button>
-          </div>
+          </>
```
Comment (bntti) [inline client/src/components/course-view/AttainmentsView.tsx:340]: You can just do `{editRights && unsavedChanges && (...`

File: client/vite.config.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@
 import react from '@vitejs/plugin-react';
 // import license from 'rollup-plugin-license';
 // import {defineConfig} from 'vitest/config';
-import {defineConfig} from 'vite';
+import { defineConfig } from 'vite';
```
Comment (bntti) [inline client/vite.config.ts:10]: This conflicts with our prettier styles. Should be `{defineConfig}`

**Review by bntti — COMMENTED** [review]
Looks good! But still a few comments
1. The A! Grades logo turns purple as it is a link
2. Edit course & grades view have scrolling
3. I would like for the linting errors to be fixed before merging
4. I would prefer if the commented code was removed
5. When selecting grades the calculate and export buttons flash momentarily
6. '2 selected student'  missing plural 'students'
7. The group by button does not look like a mui component?
8. What is the 'filters' box, it is not a button. Is it just a placeholder for the future?
9. You can select multiple group by options but it closes after selecting one, making selecting multiple tedious
10. If selecting group by latest attainment the button says 'group by grouping'
11. If multiple groupings are selected, there is no space between them
12. Grouping by student_studentNumber is something we porbably don't need
13. The side button 'grading models' wraps around to a second line, is it nicer if it was only one line?

File: client/vite.config.ts (hunk 3 above)
Comment (gubo97000) [reply client/vite.config.ts:10]: something is wrong with my vscode config? On save runs a the import fixer but not the linter??

Comment (gubo97000) [reply client/vite.config.ts:10]: I changed to include vite config in the linter. Shouldn't have broken anything 😁

**Comment by bntti** [comment]
I merged the renaming pr, you should now change `assessment model` to `grading model` and `attainment` to `course part` where applicable

**Pull request merged**